docs/proposal.md

```diff
@@ -0,0 +1,108 @@
+* **Streaming server authentication**: ([#36666](https://github.com/kubernetes/kubernetes/issues/36666)) CRI-containerd will be out-of-process with Kubelet, so it could not reuse Kubelet authentication. Its streaming server should implement its own authentication mechanism.
+* **Move container facilities into pod cgroup**: Container facilities including container image puller, container streaming handler, log handler and containerd-shim serve a specific container. They should be moved to the corresponding pod cgroup, and the overhead introduced by them should be charged to the pod.
+* **Log rotation**: ([#42718](https://github.com/kubernetes/kubernetes/issues/42718)) Container log rotation is under design. A function may be added in CRI to signal the runtime to reopen log file. CRI-containerd should implement that function after it is defined.
+* **Exec container**: With the flexibility provided by containerd, it is possible to implement `Exec` with a separate container sharing the same rootfs and mount namespace with the original container. The advantage is that the `Exec` container could have it's own sub-cgroup, so that it will not consume the resource of application container and user could specify dedicated resource for it.
+* **Advanced image management**: The image management interface in CRI is relatively simple because the requirement of Kubelet image management is not clearly scoped out. In the future, we may want to leverage the flexibility provided by containerd more, e.g. estimate image size before pulling etc.
+* ...
+## Roadmap and Milestones
+### Milestones
+#### Kubernetes 1.7 - Q2
+* [P0] Basic container lifecycle.
+* [P0] Basic image management.
+* [P0] Container networking.
+* [P1] Container streaming/logging.
+* [P2] Container/ImageFS Metrics.
+#### Kubernetes 1.8 - Q3
+* [P0] Feature complete, pass 100% cri validation test.
+* [P0] Integrate CRI-containerd with Kubernetes, and build the e2e/node e2e test framework.
+* [P1] Address the debuggability problem.
+### Q2 Roadmap
+|              Item              | 1/2 Mar. | 2/2 Mar. | 1/2 Apr. | 2/2 Apr. | 1/2 May. | 2/2 May. |
+|:------------------------------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|
+|             Survey             |     ✓    |          |          |          |          |          |
+|               POC              |          |     ✓    |          |          |          |          |
+|            Proposal            |          |          |     ✓    |          |          |          |
+|   Containerd Feature Complete  |     ✓    |     ✓    |     ✓    |          |          |          |
+| Runtime Management Integration |          |          |     ✓    |     ✓    |     ✓    |     ✓    |
+|  Image Management Integration  |          |          |          |     ✓    |     ✓    |     ✓    |
```

Comment:
Should break these last two out into the separate parts listed above in the 1.7 and 1.8 sections.